xapian-letor/include/xapian-letor/feature.h

```diff
@@ -55,19 +106,68 @@ class XAPIAN_VISIBILITY_DEFAULT Feature {
     /// Virtual destructor because we have virtual methods.
     virtual ~Feature();
 
-    /// Specify the database to use for feature building. This will be used by the Internal class.
+    /** Specify the database to use for feature building.
+     *
+     *  This will be used by FeatureList::Internal class.
+     */
     void set_database(const Xapian::Database & db);
 
-    /** Specify the query to use for feature building. This will be used by the Internal class.
-     * @param query  Xapian::Query which has to be queried
-     * @exception Xapian::InvalidArgumentError will be thrown if an empty
-     *  query is supplied
+    /** Specify the query to use for feature building.
+     *
+     *  This will be used by FeatureList::Internal class.
+     *
+     *  @param query  Xapian::Query which has to be queried
      */
     void set_query(const Xapian::Query & query);
 
-    /// Specify the document to use for feature building. This will be used by the Internal class.
+    /** Specify the document to use for feature building.
+     *
+     *  This will be used by FeatureList::Internal class.
+     */
     void set_doc(const Xapian::Document & doc);
 
+    /** Sets the termfrequency that is going to be used for
+     *  Feature building.
+     *
+     *  This is used by FeatureList::Internal while populating Statistics.
+     */
+    void set_termfreq(const std::map<std::string, Xapian::termcount> &tf);
+
+    /** Sets the inverse_doc_freq that is going to be used for
+     *  Feature building.
+     *
+     *  This is used by FeatureList::Internal while populating Statistics.
+     */
+    void set_inverse_doc_freq(const std::map<std::string, double> & idf);
+
+    /** Sets the doc_length that is going to be used for Feature building.
+     *
+     *  This is used by FeatureList::Internal while populating Statistics.
+     */
+    void set_doc_length(const std::map<std::string,
+			Xapian::termcount> & doc_len);
+
+    /** Sets the collection_length that is going to be used for
+     *  Feature building.
+     *
+     *  This is used by FeatureList::Internal while populating Statistics.
+     */
+    void set_collection_length(const std::map<std::string,
+			       Xapian::termcount> & collection_len);
+
+    /** Sets the collection_termfreq that is going to be used
+     *  for Feature building.
+     *
+     *  This is used by FeatureList::Internal while populating Statistics.
+     */
+    void set_collection_termfreq(const std::map<std::string,
+				 Xapian::termcount> & collection_tf);
+
+    /// Returns the stats needed by a subclass
+    stat_flags get_stats(){
```

Comment:
Looking good to me now, except there should be a space between `)` and `{` here.

(Will add a rule to xapian-check-patch for that).